packages/typespec-vscode/src/utils.ts

```diff
@@ -389,3 +408,36 @@ export async function loadPackageJsonFile(
   if (!packageJson) return undefined;
   return packageJson as NodePackage;
 }
+
+export async function parseOpenApi3File(filePath: string): Promise<string | undefined> {
+  try {
+    const fileContent = await readFile(filePath, "utf-8");
+    const content = JSON.parse(fileContent);
+    return content;
+  } catch (e) {
+    logger.error(`Failed to load OpenAPI3 file: ${filePath}`, [e], {
```

Comment:
please just log and return undefined and let the caller to decide whether to show notification to end user about it.